addons/pkg-npm/model-java/src/main/java/org/commonjava/indy/pkg/npm/model/PackageMetadata.java

```diff
@@ -261,9 +277,293 @@ public void setAttachments( Object attachments )
+        if ( source.getRepository() != null )
+        {
+            this.setRepository( source.getRepository() );
+            changed = true;
+        }
+        if ( source.getReadme() != null )
+        {
+            this.setReadme( source.getReadme() );
+            changed = true;
+        }
+        if ( source.getReadmeFilename() != null )
+        {
+            this.setReadmeFilename( source.getReadmeFilename() );
+            changed = true;
+        }
+        if ( source.getHomepage() != null )
+        {
+            this.setHomepage( source.getHomepage() );
+            changed = true;
+        }
+        if ( source.getBugs() != null )
+        {
+            this.setBugs( source.getBugs() );
+            changed = true;
+        }
+        if ( source.getLicense() != null )
+        {
+            this.setLicense( source.getLicense() );
+            changed = true;
+        }
+        if ( source.getAttachments() != null )
+        {
+            this.setAttachments( source.getAttachments() );
+            changed = true;
+        }
+
+        // merge maintainers list
+        Iterator maintainer = source.getMaintainers().iterator();
+        while ( maintainer.hasNext() )
+        {
+            UserInfo m = (UserInfo) maintainer.next();
+            boolean s = false;
+            Iterator snapshot = maintainers.iterator();
+            while ( snapshot.hasNext() )
+            {
+                UserInfo preExisting = (UserInfo) snapshot.next();
+                if ( preExisting.getName().equals( m.getName() ) )
+                {
+                    s = true;
+                    break;
+                }
+            }
+            if ( !s )
+            {
+                this.addMaintainers( new UserInfo( m.getName(), m.getEmail(), m.getUrl() ) );
+                changed = true;
+            }
+        }
+
+        // merge keywords list
+        Iterator keyword = source.getKeywords().iterator();
+        while ( keyword.hasNext() )
+        {
+            String key = (String) keyword.next();
+            if ( !keywords.contains( key ) )
+            {
+                this.addKeywords( key );
+                changed = true;
+            }
+        }
+
+        // merge users map
+        Map<String, Boolean> sourceUsers = source.getUsers();
+        for ( final String user : sourceUsers.keySet() )
+        {
+            if ( users.keySet().contains( user ) && users.get( user ).equals( sourceUsers.get( user ) ) )
+            {
+                continue;
+            }
+            else
+            {
+                users.put( user, sourceUsers.get( user ) );
+                changed = true;
+            }
+        }
+
+        // merge time map
+        SimpleDateFormat sdf = new SimpleDateFormat( TIMEOUT_FORMAT );
+        Map<String, String> sourceTimes = source.getTime();
+
+        Iterator timeKey = sourceTimes.keySet().iterator();
+        boolean added = false;
+
+        while ( timeKey.hasNext() )
```

Comment:
@jdcasey I tweak this block based on some of your comment, see the new commit pls. And seem there are some confusing caused by the param naming. I change that. Time map is the combination like <versionName, updateTime> and sorting as its values --> time, not keys.